pkg/loki/modules.go

```diff
@@ -507,7 +507,7 @@ func (t *Loki) initQuerier() (services.Service, error) {
 	svc, err := querier.InitWorkerService(
 		querierWorkerServiceConfig,
 		prometheus.DefaultRegisterer,
-		handler,
+		serverutil.RecoveryMiddleware.Wrap(handler),
```

Comment:
double checking: shouldn't you wrap `httpHandler` here instead of `handler`?

Reply:
probably worth clarifying this via a comment somewhere that's easy to find, my understanding is this;
- there's now the base level querier API, which supports protobuf
- on top of that we use a wrapper function to turn that APIs responses into the currently used http responses (httpgrpc)

so I think the same question is likely to come up again, and we should make it obvious when a middleware needs to be added to both the `handler` and `httpHandler` vs just one of the two

Reply:
Should I rename the handler? I can add some documentation.

Reply:
Lets rename or add docs/comments when you're back